A36-xds-for-servers.md

```diff
@@ -0,0 +1,401 @@
+to be invoked when the server begins serving and when the server encounters
+errors that force it to be "not serving". If "not serving", the callback must be
+provided error information, for debugging use by developers. The error
+information should generally be language-idiomatic, but determining the cause of
+the error does not need to be machine-friendly. If the application does not
+register the callback, XdsServer should log any errors and each serving
+resumption after an error, all at a default-visible log level.
+
+XdsServer's start must not fail due to transient xDS issues, like missing xDS
+configuration from the xDS server. If XdsServer's start blocks waiting for xDS
+configuration an application can use the serving state callback to be notified
+of issues preventing startup progress.
+
+### xDS Protocol
+
+The `GRPC_XDS_BOOTSTRAP` file will be enhanced to have a new field:
+```
+{
+  // A template for the name of the Listener resource to subscribe to for a gRPC
+  // server. If the token `%s` is present in the string, it will be replaced
+  // with the server's listening "IP:port" (e.g., "0.0.0.0:8080", "[::]:8080").
+  "server_listener_resource_name_template": "example/resource/%s",
+  // ...
+}
+```
+
+XdsServer will use the normal `XdsClient` to communicate with the xDS server.
+There is no default value for `server_listener_resource_name_template` so if it
+is not present in the bootstrap then server creation or start will fail or the
+XdsServer will become "not serving". XdsServer will perform the `%s` replacement
+in the template (if the token is present) to produce a listener name. The
+XdsServer will start an XdsClient watch on the listener name for a
+`envoy.config.listener.v3.Listener` resource. No special character handling of
+the template or its replacement is performed. For example, with an address of
+`[::]:80` and a template of `grpc/server?xds.resource.listening_address=%s`, the
+resource name would be `grpc/server?xds.resource.listening_address=[::]:80`.
+
+To be useful, the xDS-returned Listener must have an
+[`address`][Listener.address] that matches the listening address provided. The
+Listener's `address` would be a TCP `SocketAddress` with matching `address` and
+`port_value`. The XdsServer must be "not serving" if the address does not match.
+
+The xDS client must NACK the Listener resource if `Listener.listener_filters` is
+non-empty.
+
+### FilterChainMatch
+
+Although `FilterChain.filters`s will not be used as part of this gRFC, the
+`FilterChain` contains data that may be used like TLS configuration in
+`transport_socket`. When looking for a FilterChain, the standard matching logic
+must be used. The [most specific `filter_chain_match`][matching logic] of the
+repeated [`filter_chains`][Listener.filter_chains] should be found for the
+specific connection and if one is not found, the `default_filter_chain` must be
+used. The following is a snippet of [all current fields of
```

Comment:
If no `default_filter_chain` is present, the connection is closed.

Reply:
Was that a question or a statement? If that's a question, I'd like to know the answer for the same.

Reply:
It's a statement, potentially serving as the start of a discussion if we want to do something else when there are are no filter chains that we can use.